### ISSUE TITLE
Constantly spin around in circles

### ISSUE THREAD

**Issue by dborosev** [mdovgialo/steam-vr-wheel#10]
Using the gamepad mode, and all it does is spin around in circles.  If I just use a xbox controller it works fine, but with the vive controllers bat file, it spins and cannot play any games.
Had it working a little up until today when it decided to just spin.  Tried restarting games, etc.

**Comment by mdovgialo** [comment]
Have you tried going to - Control panel -> USB Game Controllers -> vjoy and then clicking "Calibrate" and following the on screen instructions?

Does your touchpad work properly in Vive games?

**Comment by dborosev** [comment]
Yeah my xbox controller and vive wands both work fine on their own for Vive games.  Its weird because this seemed to work fine the first time I installed and tried it, and now all I get are spinning in circles whenever I use the gamepad.bat file.  

When I go into the vjoy settings, the buttons work and show as being pressed, but the trackpad on the vive does not move the crosshairs.  So I cannot calibrate it, as it won't move.

**Comment by dborosev** [comment]
Also tried uninstalling vjoy, reinstalling, resetting, rebooting, etc.
Still just spins around in circles.  The xbox controller works fine on it own.

**Comment by mdovgialo** [comment]
Have you tried running the windows USB Game Controller calibration wizard for vJoy anyway?
I'll look what happens with my gamepad mode as well, later today, or tomorrow.

**Comment by dborosev** [comment]
Yeah, I ran that calibration, and I could not get anywhere with it because the trackpad on the vive doesn't move anything.  The Vive buttons work and show up, but no axis movement.
It did work the first time I installed and tried it.  Then the next day, it just started spinning, nothing changed by me.

**Comment by mdovgialo** [comment]
One more question - which steam-vr-wheel version are you using?

**Comment by dborosev** [comment]
2.5.2

**Comment by mdovgialo** [comment]
Interesting, there is no problem on my side. I'm running latest Windows 10, vjoy, stable Steam VR, and steam-vr-wheel. What is your configuration?

Try running steam-vr-gamepad-debug.bat from "DEBUG_MODES" folder and report if something changes when you touch touchpad.

**Comment by dborosev** [comment]
My config is the same as yours.  Everything works independently, but with vjoy/wheel it just spins around.  And no way to see any of this in the vjoy config as the trackpad doesn't show any axis movement on it.
I'll try debug and see what I get.

**Comment by dborosev** [comment]
Debug reports back the proper inputs.  axis movement, button presses.  But even in debug mode, the game still just spins.

**Comment by dborosev** [comment]
Found this issue:
  https://steamcommunity.com/app/264710/discussions/0/154641879453287770/

Seems to be related.  Same game even.  But even when I try disabling my xbox controller, it still fails.  

Seems like vjoy doesn't play well with certain games?  Maybe you can understand what is going on better than me.
 The WEIRDEST thing is that it WAS definitely working fine the first time I tried it.  The only thing I did was reboot.  Maybe the reboot is what caused it.

**Comment by dborosev** [comment]
Any thoughts?

**Comment by mdovgialo** [comment]
I'm really at loss. I cannot recreate it.

Hmmm. When you go into the USB game controller -> vJoy -> calibration,
![image](https://user-images.githubusercontent.com/6193380/37572828-4d01b47c-2b11-11e8-86ec-56faa3842c4c.png)
And on this screen enable the "Show raw data" checkbox:
![image](https://user-images.githubusercontent.com/6193380/37572833-5f3ecdd2-2b11-11e8-8258-bcc5aefe2fe2.png)
Do the numbers change? When you touch any of the trackpads?
You should also try to go through whole calibration process.

Do you have any other controllers enabled? Did you enable all axis in vJoy configurator?

**Comment by dborosev** [comment]
I'm confused, are you telling me to do this with the vive controllers while running your .bat file?  If so, then they do nothing in the vjoy calibration.  (not xbox controler callibration)   However I see the axis numbers printed out in your debug bat file when I use the vive trackpads, so there is movement registered there.
Might be worth mentioning I'm trying this with subnautica, and according to the link I sent you, unity games may have issues with vjoy?  Not sure if thats true.

**Comment by mdovgialo** [comment]
If touchpad is not getting mapped to the vJoy axis then it is a more fundamental problem than unity not working with vJoy (which I highly doubt). Steam-VR-wheel properly detects you touching the gamepad, so the problem must be somewhere in the vJoy.


I'm really confused about your problem, and really don't understand why does that happen.
 Do the other .bat files work? For example, when you move the virtual wheel, do the vJoy controllers axis move?

**Comment by dborosev** [comment]
The first thing I tried when I got it was project cars.  The wheel turned then.   I just tried it again, and the wheel appears, and I can move the wheel around, but I can't steer it, even when gripping.  
It just floats there.  

When I run the calibrate for vjoy device, the trackpad doesn't respond.  The buttons work, as it goes to the next option, but the axis is stuck in the top left corner of the box.  No movement.

I reinstalled vjoy again.  Same thing.  Worth noting that in the vjoy config, the buttons on the vive work fine, its just the trackpad that doesn't respond..
